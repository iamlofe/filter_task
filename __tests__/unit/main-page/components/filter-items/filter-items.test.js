import React from 'react';
import { shallow } from 'enzyme';
import { List } from 'immutable';

import FilterItems from 'main-page/components/filter-items/filter-items';
import FilterCheckbox from 'main-page/components/filter-checkbox/filter-checkbox';
import { getMockListItems } from '../../mock-store/mock-filter-reducer';

describe('account component', () => {
    const baseProps = {
        filteredList: new List(),
        selectedItems: new List(),
        filterId: '',
        onChangeState: jest.fn(),
        displayClass: ''
    };

    it('is shallow rendered without crashing', () => {
        const filterChecbox = <FilterItems {...baseProps} />;

        const filterChecboxComponent = shallow(filterChecbox);
        expect(filterChecboxComponent).toMatchSnapshot();
    });

    it('is shallow rendered with not empty filteredList without crashing', () => {
        const filterChecbox = <FilterItems {...baseProps} filteredList={getMockListItems()} />;

        const filterChecboxComponent = shallow(filterChecbox);
        expect(filterChecboxComponent).toMatchSnapshot();
    });

    it('should truth length', () => {
        const filterChecbox = <FilterItems {...baseProps} />;

        const filterChecboxComponent = shallow(filterChecbox);
        expect(filterChecboxComponent.find(FilterCheckbox).length).toEqual(0);
    });
});
