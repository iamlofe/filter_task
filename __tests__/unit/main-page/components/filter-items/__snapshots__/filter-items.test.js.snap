// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`account component is shallow rendered with not empty filteredList without crashing 1`] = `
<Fragment>
  <FilterCheckbox
    checked={false}
    displayClass=""
    filterId=""
    filteredItem={
      Immutable.Record {
        "id": 3,
        "title": "results",
        "contextId": 1,
        "demisionId": 2,
        "resultId": 3,
      }
    }
    key="3"
    onChangeState={[MockFunction]}
    selectedItems={Immutable.List []}
  />
  <FilterCheckbox
    checked={false}
    displayClass=""
    filterId=""
    filteredItem={
      Immutable.Record {
        "id": 4,
        "title": "results2",
        "contextId": 1,
        "demisionId": 2,
        "resultId": 4,
      }
    }
    key="4"
    onChangeState={[MockFunction]}
    selectedItems={Immutable.List []}
  />
</Fragment>
`;

exports[`account component is shallow rendered without crashing 1`] = `<Fragment />`;
